newIDE/app/src/Export/GenericExporters/OnlineWebExport.js

```diff
@@ -56,20 +59,42 @@ export const WebProjectLink = ({ build, loading }: WebProjectLinkProps) => {
   };
 
   return (
-    <Line justifyContent="center">
-      <TextField value={value} readOnly fullWidth />
-      {!buildPending && (
-        <>
-          <RaisedButton label={<Trans>Open</Trans>} onClick={onOpen} />
-          <RaisedButton primary icon={<Paste />} onClick={onCopy} />
-        </>
-      )}
+    <>
+      <TextFieldWithButtonLayout
+        noFloatingLabelText
+        renderTextField={() => (
+          <TextField
+            value={value}
+            readOnly
+            fullWidth
+            endAdornment={
+              <IconButton
+                disabled={!!buildPending}
+                onClick={() => onCopy()}
```

Comment:
this can just be `onCopy` - it's probably not important any more, but I think with React Classes, this used to cause a re-render because the function was re-created every time?
Anyway, I think it's more clean 😇

Reply:
This won't help re-renders in this particular case (because onCopy is not a React.useCallback) BUT you're totally right that we should strive to have as few functions as possible (especially because with this, we're not far from being able to entirely switch IconButton to a pure component/React.memo and avoid a bunch of useless renders!)

Fixed :)